### DISCUSSION TITLE
Creating mutable array from references?

### DISCUSSION THREAD

**Discussion by Arty-Maly** [jorgecarleitao/arrow2#1315] [Q&A] [answered]
Hi, I am curious if there is a way to create a MutablePrimitiveArray from a section of values from another MutablePrimitiveArray? 

They way I understand it, I will always incur a copy cost? There would be no way to have a zero copy in this situation? 
I was hoping to pass a ref of the values from one array to another.

**Comment by sundy-li** [comment, marked as answer]
`MutablePrimitiveArray ` need to own `Vec<T>`, so if you want to create it from another MutablePrimitiveArray, you need to move the memory otherwise it will have copy cost.

**Reply by Arty-Maly** [reply]
Ok. Thanks for your quick response. 

One workaround I thought of was to convert MutablePrimitiveArray to PrimitiveArray (which is O(1)) and then use PrimitiveArrays with the slice method. That would give me a zero copy as I understand it, correct?

One downside is that I lose the mutability while I have the extra slices in use. (Can’t conver back to mutable array)

**Reply by jorgecarleitao** [reply]
Yes, you can think of them as `struct MutablePrimitiveArray(Vec<T>)` and `struct PrimitiveArray(Arc<Vec<T>>, offset, length)` where `offset` and `length` correspond to the slice. This is why `PrimitiveArray` has a `O(1)` cost of slicing.

**Reply by Arty-Maly** [reply]
Thanks for helping me understand.
